CI: Integration Tests: operator/watchers TestScript failed: found unexpected goroutines

## CI failure

https://github.com/cilium/cilium/actions/runs/16005973989/job/45152635248#step:15:112
From PR https://github.com/cilium/cilium/pull/40286, which is unrelated to this test case.

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/operator/watchers   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestScript (0.47s)

    logger.go:256: time=2025-07-01T17:31:37.718Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:357 msg="Starting hive"
    logger.go:256: time=2025-07-01T17:31:37.718Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:361 msg="Started hive" duration=57.619µs
    logger.go:256: time=2025-07-01T17:31:37.823Z level=INFO source=/home/runner/work/cilium/cilium/pkg/k8s/resource_ctors.go:363 msg="Using discoveryv1.EndpointSlice" module=k8s-resources
    logger.go:256: time=2025-07-01T17:31:38.024Z level=INFO source=/home/runner/work/cilium/cilium/operator/watchers/service_sync.go:147 msg="Initial list of services successfully received from Kubernetes" module=service-sync
    logger.go:256: time=2025-07-01T17:31:38.528Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:373 msg="Stopping hive"
    logger.go:256: time=2025-07-01T17:31:38.529Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:377 msg="Stopped hive" duration=326.569µs
    script_test.go:43: found unexpected goroutines:
        [Goroutine [113](https://github.com/cilium/cilium/actions/runs/16005973989/job/45152635248#step:15:114) in state chan receive, with k8s.io/client-go/util/workqueue.(*Typed[...]).updateUnfinishedWorkLoop on top of the stack:
        k8s.io/client-go/util/workqueue.(*Typed[...]).updateUnfinishedWorkLoop(0x3e111c0)
        	/home/runner/work/cilium/cilium/vendor/k8s.io/client-go/util/workqueue/queue.go:355 +0x92
        created by k8s.io/client-go/util/workqueue.newQueue[...] in goroutine 84
        	/home/runner/work/cilium/cilium/vendor/k8s.io/client-go/util/workqueue/queue.go:176 +0x1e7
         Goroutine [117](https://github.com/cilium/cilium/actions/runs/16005973989/job/45152635248#step:15:118) in state chan receive, with k8s.io/client-go/util/workqueue.(*Typed[...]).updateUnfinishedWorkLoop on top of the stack:
        k8s.io/client-go/util/workqueue.(*Typed[...]).updateUnfinishedWorkLoop(0x3e111c0)
        	/home/runner/work/cilium/cilium/vendor/k8s.io/client-go/util/workqueue/queue.go:355 +0x92
        created by k8s.io/client-go/util/workqueue.newQueue[...] in goroutine 84
        	/home/runner/work/cilium/cilium/vendor/k8s.io/client-go/util/workqueue/queue.go:176 +0x1e7
```

⛑️ The following owners are responsible for reliability of the failing test(s): 
Package                                      Owner
github.com/cilium/cilium/operator/watchers   @cilium/operator

--- a/operator/watchers/script_test.go
+++ b/operator/watchers/script_test.go
@@ -38,9 +38,18 @@ import (
 var debug = flag.Bool("debug", false, "Enable debug logging")
 
 func TestScript(t *testing.T) {
-	// Catch any leaked goroutines. Ignoring goroutines possibly left by other tests.
-	leakOpts := goleak.IgnoreCurrent()
-	t.Cleanup(func() { goleak.VerifyNone(t, leakOpts) })
+	// Catch any leaked goroutines.
+	t.Cleanup(func() {
+		goleak.VerifyNone(t,
+			// Ignore goroutines possibly left by other tests.
+			goleak.IgnoreCurrent(),
+
+			// Ignore goroutine started by the workqueue. It reports metrics
+			// on unfinished work with default tick period of 0.5s - it terminates
+			// no longer than 0.5s after the workqueue is stopped.
+			goleak.IgnoreTopFunction("k8s.io/client-go/util/workqueue.(*Type).updateUnfinishedWorkLoop"),
+		)
+	})
 
 	version.Force(testutils.DefaultVersion)
 

